### PR TITLE
Add XQuery mode

### PR DESCRIPTION
**Pull request by wcandillon**

### CODE REVIEW

**Comment by busykai** [comment]
@wcandillon, you need to sign [Contributor License Agreement](http://dev.brackets.io/brackets-contributor-license-agreement.html) before your contribution can be accepted.

**Comment by wcandillon** [comment]
@busykai did it work? I got the following error: https://www.dropbox.com/s/skc4b4484jqdsuw/Screenshot%202015-01-03%2018.30.27.png?dl=0

Also, the way JSONiq is supported in c9.io and atom.io is by plugin XQLint in the IDE API. For instance: https://atom.io/packages/language-jsoniq
Could you point me out how to do it for brackets?

I made a PR for a JSONiq mode in CodeMirror which got closed a bit abruptly: https://github.com/codemirror/CodeMirror/pull/3014

Are you guys happy with CoreMirror? I've been a ACE contributor and now dabbling with Atom and I find CoreMirror a bit too primitive.

**Comment by MarcelGerber** [comment]
Yeah, it worked, the build passed now, but the CLA submit page should be fixed nonetheless.

I can understand Marijn, you don't need a linter with [an array of 37000 random-looking numbers](https://github.com/codemirror/CodeMirror/blob/ff4b6fb2d48b252bcfb53a71899ca97646819907/mode/jsoniq/jsoniq.js#L1645-L3753) for the "simple" job of syntax-highlighting. That is, for syntax highlighting (as opposed to linting), there's no need to be aware of every problem in the code.

Take a look at the [XQuery mode](https://github.com/codemirror/CodeMirror/blob/master/mode/xquery/xquery.js), too. That's how a normal CM mode looks like.
And if JSONiq is like JSON/JSON-LD, you can also extend the [JS mode](https://github.com/codemirror/CodeMirror/blob/master/mode/javascript/javascript.js).
As you can see, no other mode is based on a linter, and thus the other modes are usually around 300-800 lines.

File: src/language/languages.json [hunk 1]
```diff
@@ -205,6 +205,12 @@
         "fileExtensions": ["sql"]
     },
 
+    "xquery": {
+        "name": "XQuery",
+        "mode": ["xquery", "application/xquery"],
+        "fileExtensions": ["xq", "jq"]
+    },
```
Comment (MarcelGerber) [inline src/language/languages.json]: Please add `"blockComment": ["(:", ":)"]`

**Comment by redmunds** [comment]
@wcandillon For modes that are not in core CodeMirror, you can programmatically define them in a Brackets extension. Take a look at the [Brackets Apache extension](https://github.com/tpryan/Brackets-Apache/blob/master/main.js) as an example.

**Comment by wcandillon** [comment]
Awesome @redmunds thank you so much for the pointer.

**Comment by wcandillon** [comment]
@redmunds How can you define and import a javascript module in brackets? Adding an npm dependency and using `require` doesn't seem to work for bracket extensions.

**Comment by MarcelGerber** [comment]
@wcandillon Just create a new folder in the directory shown after clicking `Help > Show Extensions Folder`. The `main.js` in that newly created folder is automatically loaded on Brackets startup. You can require Brackets' modules using `brackets.getModule()` and `require()` your extensions files the normal way.

Notice though that you also need a `package.json` for your extension.

[Wiki: How to write extensions](https://github.com/adobe/brackets/wiki/How-to-write-extensions)

**Comment by wcandillon** [comment]
@MarcelGerber It doesn't look that the package.json file of a bracket extension cannot contain dependencies. When I install my extension, the dependencies doesn't seem to be installed. I looked how the jshint extension does it (jshint is committed within the repo). I guess I will try the same.

**Comment by MarcelGerber** [comment]
Yeah, dependency management is not something we currently have in Brackets.
You have to at least include the dependency files in the .zip files you submit to the Brackets extension registry later, there's no need to have them in your actual repo (that is, you can `.gitignore` them, for example).

**Comment by wcandillon** [comment]
@MarcelGerber do you have a plugin example I could look at? `require` doesn't seem to work for me.

**Comment by MarcelGerber** [comment]
You can look at this, for example: https://github.com/adobe/brackets/blob/master/src/extensions/default/WebPlatformDocs/main.js#L39

Btw, you might get faster support on IRC: Freenode.net/#brackets

**Comment by ficristo** [comment]
@wcandillon thank you for your contribution.
As of now we don't accept new languages.

**Pull request closed without merging**